crates/e2e-testing/src/spin_controller.rs

```diff
@@ -46,6 +49,22 @@ impl Controller for SpinUp {
     ) -> Result<AppInstance> {
         let appdir = spin::appdir(app_name);
 
+        //cleanup state dir before spin up
+        let state_dir = match env::var("STATE_DIR") {
+            Ok(val) => val,
+            Err(_) => Path::new(appdir.as_str())
+                .join(".spin")
+                .into_os_string()
+                .into_string()
+                .unwrap(),
+        };
+        if fs::remove_dir_all(&state_dir).is_err() {
+            println!(
+                "warning: failed to cleanup existing statedir {}",
+                &state_dir
+            )
+        };
+
         let mut cmd = vec!["spin", "up"];
```

Comment:
Do you want to add `"--state-dir", state_dir` here?

Reply:
I tried adding a tempdir here, but it cleans up before we had opportunity to run the actual tests (as tempdir goes out of scope)

Reply:
The way you're doing it makes the TempDir a temporary value, which allows it to drop immediately, which in turn cleans up the temporary directory.  Could you try something like:

```
let temp_dir = TempDir::new().unwrap();
let state_dir = format!("{}", temp_dir.path().display());
// party on state_dir
tc.run(controller).await.unwrap();
drop(temp_dir);
```

and see if that helps?

(The explicit `drop` is optional: `temp_dir` will get dropped at the end of the scope anyway.  I quite like it because it makes it clear when we expect the drop to happen (on the happy path), and provides a compiler error if someone later "improves" the code in ways that would make `temp_dir` go out of state too early.  Others disagree though!)

Reply:
so, I am not adding a temp dir, but adding the state-dir to be test app dir explicitly (unless the flag is provided by user). this will also mean our testcase will ignore STATE_DIR env variable